src/AbstractWebApplication.php

```diff
@@ -788,10 +788,15 @@ protected function isRedirectState($state)
 	public function isSslConnection()
 	{
 		$serverSSLVar = $this->input->server->getString('HTTPS', '');
+
+		if (!empty($serverSSLVar) && strtolower($serverSSLVar) !== 'off')
+		{
+			return true;
+		}
+
 		$serverForwarderProtoVar = $this->input->server->getString('HTTP_X_FORWARDED_PROTO', '');
 
-		return (!empty($serverSSLVar) && strtolower($serverSSLVar) != 'off') ||
-		(!empty($serverForwarderProtoVar) && strtolower($serverForwarderProtoVar) == 'https');;
+		return !empty($serverForwarderProtoVar) && strtolower($serverForwarderProtoVar) === 'https';
```

Comment:
serverSSLVar isn't used anymore in the return now. This should be reverted.

Reply:
I changed it so instead of one complex if check it first checks the `$serverSSLVar` value and returns early, falling back to checking the `$serverForwarderProtoVar` value.  Just makes it a little easier to read and follow.

Reply:
You are right, I missed the first return true statement. Thanks!